cmd/extensions/oidc-policy/internal/controller/oidcpolicy_reconciler.go

```diff
@@ -138,29 +140,31 @@ func (r *OIDCPolicyReconciler) Reconcile(ctx context.Context, request reconcile.
 	return reconcile.Result{}, nil
 }
 
-func (r *OIDCPolicyReconciler) reconcileSpec(ctx context.Context, pol *kuadrantv1alpha1.OIDCPolicy, igw *ingressGatewayInfo) error {
+func (r *OIDCPolicyReconciler) reconcileSpec(ctx context.Context, pol *kuadrantv1alpha1.OIDCPolicy, igw *ingressGatewayInfo) (*kuadrantv1.AuthPolicy, error) {
 	// Reconcile AuthPolicy for the oidc policy http route
-	if err := r.reconcileMainAuthPolicy(ctx, pol, igw); err != nil {
+	mainAuthPol, err := r.reconcileMainAuthPolicy(ctx, pol, igw)
+	if err != nil {
 		r.Logger.Error(err, "Failed to reconcile main AuthPolicy")
-		return err
+		return nil, err
 	}
 
 	// Reconcile HTTPRoute for the callback for exchanging code/token
-	if err := r.reconcileCallbackHTTPRoute(ctx, pol, igw); err != nil {
+	if err = r.reconcileCallbackHTTPRoute(ctx, pol, igw); err != nil {
 		r.Logger.Error(err, "Failed to reconcile callback HTTPRoute")
-		return err
+		return nil, err
 	}
 	// Reconcile AuthPolicy for the Token exchange flow with metadata http call
-	if err := r.reconcileCallbackAuthPolicy(ctx, pol, igw); err != nil {
+
+	if err = r.reconcileCallbackAuthPolicy(ctx, pol, igw); err != nil {
 		r.Logger.Error(err, "Failed to reconcile callback AuthPolicy")
-		return err
+		return nil, err
 	}
 
-	return nil
+	return mainAuthPol, nil
```

Comment:
Doesn't the status depend on all of these things being in a good state? why is only the main auth policy checked?

Reply:
True, was just lazy condition hehe. Will update